multiqc/modules/haplocheck/haplocheck.py

```diff
@@ -0,0 +1,137 @@
+import csv
+import logging
+from collections import OrderedDict
+
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.plots import bargraph, table
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="Haplocheck",
+            anchor="haplocheck",
+            href="https://github.com/genepi/haplocheck",
+            info=(
+                "Haplocheck detects contamination patterns in mtDNA AND WGS sequencing studies by analyzing "
+                "the mitochondrial DNA."
+            ),
+            doi="10.1101/gr.256545.119",
+        )
+        log_files = self.find_log_files("haplocheck", filehandles=True)
+        samples = {}
+        for f in log_files:
+            samples = self._parse_samples(f, samples)
+            self.add_data_source(f)
+
```

Comment:
Please add a `self.ignore_samples` call after parsing (see [docs](https://multiqc.info/docs/development/modules/#filtering-by-parsed-sample-names)), followed by a `raise UserWarning` if no samples are left in the dictionary ([docs](https://multiqc.info/docs/development/modules/#no-files-found)).

https://github.com/ewels/MultiQC/blob/master/multiqc/modules/samblaster/samblaster.py#L34-L37


https://multiqc.info/docs/development/modules/#filtering-by-parsed-sample-names